### PR TITLE
Configure Continuous Delivery for frontend

### PR DESCRIPTION
**Pull request by BenCoffeed**
- [x] Add the Clubhouse Story ID: [ch20603]
- [x] Add QA tasks for anything needing testing _not_ behind a flag
- [x] Update or add any QA tasks on the feature flag card

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,24 +2,65 @@ version: 2.1
...
+          name: move dist directory into built_code directory
+          command: mv ./dist built_code/
+      - persist_to_workspace:
+          root: ./built_code
+          paths:
+            - dist
   deploy:
-    executor: cypress/base-10
+    docker:
+      - image: 'circleci/python:2.7'
```
Comment (paulfalgout) [inline .circleci/config.yml]: I don't think this orb requires python, it only lists bash.. is there any reason not to use the previous environment?  It'll be a lot faster as you won't have to run npm install or deal with using caches

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,24 +2,65 @@ version: 2.1
...
     steps:
       - attach_workspace:
-          at: ~/
-      - run: npm run build
-      - run: echo S3 orb stuff for dist/
```
Comment (paulfalgout) [inline .circleci/config.yml]: I _kinda_ think you could drop all of the build steps added and just add this at line 12:
```yml
  - aws-s3/sync:
      from: /dist
      to: s3://$bucket_name
      overwrite: true
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -2,24 +2,65 @@ version: 2.1
...
+          name: Authenticate with registry
+          command: 'npm config set "//npm.fontawesome.com/:_authToken" $FONTAWESOME'
+      - run:
+          name: Install
+          command: npm i --no-dev
+      - run: 
+          name: Run build script
+          command: 'npm run build'
+      - save_cache:
+          key: npm-{{ checksum "composer.lock" }}
```
[the remark this answers is not in the document]
Comment (BenCoffeed) [reply .circleci/config.yml]: ha! Copy pasta caught!

File: .circleci/config.yml [hunk 4]
```diff
@@ -2,24 +2,65 @@ version: 2.1
...
+            - npm-
+      - run:
+          name: Set registry URL
+          command: 'npm config set "@fortawesome:registry" https://npm.fontawesome.com/'
+      - run:
+          name: Authenticate with registry
+          command: 'npm config set "//npm.fontawesome.com/:_authToken" $FONTAWESOME'
+      - run:
+          name: Install
+          command: npm i --no-dev
```
[the remark this answers is not in the document]
Comment (BenCoffeed) [reply .circleci/config.yml]: So, does ci include dev dependencies?

File: .circleci/config.yml (hunk 1 above)
Comment (BenCoffeed) [reply .circleci/config.yml]: AWS CLI needs python.

File: .circleci/config.yml (hunk 2 above)
Comment (BenCoffeed) [reply .circleci/config.yml]: If we reuse built items, most likely, yes.

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (paulfalgout) [reply .circleci/config.yml]: yes, but that is a good thing.  the dev dependencies are what does the building, the regular dependency are shipped with the build.

File: .circleci/config.yml (hunk 2 above)
Comment (paulfalgout) [reply .circleci/config.yml]: well I mean the build runs on line 11.  I don't see a need to create a new environment to do the build, the results from line 11 and the new build should be identical.

Comment (paulfalgout) [reply .circleci/config.yml]: well I mean the build runs on line 11.  I don't see a need to create a new environment to do the build.. the results from line 11 and the new build should be identical.

Comment (BenCoffeed) [reply .circleci/config.yml]: 👍 Alsot, unforunately, the aws-s3 orb from CircleCI only includes commands and not jobs. So, it still needs to be run in a defined job context.

File: .circleci/config.yml (hunk 1 above)
Comment (BenCoffeed) [reply .circleci/config.yml]: See note about aws-s3 orb not having jobs defined, but only commands below.

File: .circleci/config.yml [hunk 5]
```diff
@@ -2,24 +2,65 @@ version: 2.1
...
+  build:
+    docker: 
+      - image: circleci/node:10
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - npm-{{ checksum "package-lock.json" }}
+            - npm-
+      - run:
```
[the remark this answers is not in the document]
Comment (BenCoffeed) [reply .circleci/config.yml]: If you're comfortable with shipping what is built in the ci workflow, I'm all for not building another time.

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (BenCoffeed) [reply .circleci/config.yml]: 👍

**Comment by BenCoffeed** [comment]
CodeCov is complaining. I queued up a re-run since the issues reported have nothing to do with the single file I changed.

**Comment by BenCoffeed** [comment]
Yes. I intended on removing in a separate PR since the inclusion of that is really a testing step for this PR.

**Comment by paulfalgout** [comment]
I think it's safe to update since the test is recorded here https://circleci.com/workflow-run/1a13cffa-1149-4d70-b880-eee2001d0143 ?  But we can run a 2nd PR if you'd prefer

**Pull request merged**